protobuf/v1/nexus.proto

```diff
@@ -112,6 +112,7 @@ message Child {
   ChildStateReason state_reason = 3;  // child state reason
   int32 rebuild_progress = 4;         // rebuild progress
   optional string device_name = 5;    // child device name
+  google.protobuf.Timestamp fault_timestamp = 6;   // timestamp(UTC) when child went faulted
```

Comment:
Should it not be optional as we saw in confluence?

Reply:
Yes, it is an Option in the data-plane code. Here in the API, the autogenerated code corresponding to this is an Option as well. I'm not sure if we need `optional` keyword in protobuf file.

`pub fault_timestamp: ::core::option::Option<::prost_wkt_types::Timestamp>`

Reply:
I tried putting `optional` keyword. Though it compiles, I also stumbled upon this link where proto3 claims to have done away with `required` and `optional` field support. 
https://github.com/protocolbuffers/protobuf/issues/2497#issuecomment-267422550

Reply:
Ohh. I see. Thanks for explaining.

Reply:
Actually that's not true anymore, optional is back with different semantics IIRC related to using the default value of a field when not set.